promql/engine.go

```diff
@@ -1158,10 +1158,11 @@ func (ev *evaluator) eval(expr parser.Expr) parser.Value {
 				// Only buffer stepRange milliseconds from the second step on.
 				it.ReduceDelta(stepRange)
 			}
+			points = points[:0]
```

Comment:
It's a bit weird to do this after rather than before, it makes the logic less obvious.